programs/tests/mt-c-tests/uart/uart.c

```diff
@@ -0,0 +1,236 @@
+/**
+ * @author Magnus Mæhlum (magnmaeh@stud.ntnu.no)
+ * @brief This test implements a Software-defined UART, which is a more formal
+ *        name for a bit-banged UART.
+ * 
+ */
+
+#include <stdlib.h>
+#include <stdint.h>
+#include <errno.h>
+#include <string.h>
+#include <flexpret.h>
+
+#define CLOCKFREQ     ((int)(100e6)) // 100 MHz
+#define NS_PER_CLK    (10)
+#define UART_BAUDRATE (38400)
+#define CLKS_PER_BAUD ((int) ((CLOCKFREQ) / (UART_BAUDRATE)))
+#define NS_PER_BAUD   ((CLKS_PER_BAUD) * (NS_PER_CLK))
+
+enum State {
+    STATE_STARTBIT,
+    STATE_DATABITS,
+    STATE_STOPBIT,
+} state = STATE_STARTBIT;
```

Comment:
This is new syntax to me, what does this assignment mean?

Reply:
It is equivalent to:

```
enum State {
    STATE_STARTBIT,
    STATE_DATABITS,
    STATE_STOPBIT,
};
enum State state = STATE_STARTBIT;
```

Just more short-hand syntax, I guess :) It can be done to structs as well, I think.